MetadataMigration/src/test/java/org/opensearch/migrations/MultiTypeMappingTransformationTest.java

```diff
@@ -0,0 +1,301 @@
+            CompletableFuture.allOf(
+                    CompletableFuture.runAsync(indexCreatedCluster::start),
+                    CompletableFuture.runAsync(upgradedSourceCluster::start),
+                    CompletableFuture.runAsync(targetCluster::start)
+            ).join();
+
+            var indexCreatedOperations = new ClusterOperations(indexCreatedCluster.getUrl());
+            var upgradedSourceOperations = new ClusterOperations(upgradedSourceCluster.getUrl());
+            var targetOperations = new ClusterOperations(targetCluster.getUrl());
+
+            // Test data
+            var originalIndexName = "test_index";
+
+            // Create index and add a document on the source cluster
+            indexCreatedOperations.createIndex(originalIndexName);
+            indexCreatedOperations.createDocument(originalIndexName, "1", "{\"field1\":\"My Name\"}", null, "type1");
+            indexCreatedOperations.createDocument(originalIndexName, "2", "{\"field2\":123}", null, "type2");
+            indexCreatedOperations.createDocument(originalIndexName, "3", "{\"field3\":1.1}", null, "type3");
+
+            var arguments = new MigrateOrEvaluateArgs();
+
+            // Use SnapshotImage as the transfer medium
+            var snapshotName = "initial-setup-snapshot";
+            var snapshotContext = SnapshotTestContext.factory().noOtelTracking();
+            var sourceClient = new OpenSearchClient(ConnectionContextTestParams.builder()
+                    .host(indexCreatedCluster.getUrl())
+                    .insecure(true)
+                    .build()
+                    .toConnectionContext());
+            var snapshotCreator = new FileSystemSnapshotCreator(
+                    snapshotName,
+                    sourceClient,
+                    SearchClusterContainer.CLUSTER_SNAPSHOT_DIR,
+                    List.of(),
+                    snapshotContext.createSnapshotCreateContext()
+            );
+
+            // Get snapshot for ES 5
+            SnapshotRunner.runAndWaitForCompletion(snapshotCreator);
+            indexCreatedCluster.copySnapshotData(localDirectory.toString());
+
+            // Snapshot is automatically visible due to container mount
+
+            // Register snapshot repository
+            upgradedSourceOperations.createSnapshotRepository(SearchClusterContainer.CLUSTER_SNAPSHOT_DIR, snapshotCreator.getRepoName());
+
+            // Restore snapshot
+            upgradedSourceOperations.restoreSnapshot(snapshotCreator.getRepoName(), snapshotCreator.getSnapshotName());
+
+            // Verify that the index exists on the upgraded cluster
+            var checkIndexUpgraded = upgradedSourceOperations.get("/" + originalIndexName);
+            assertThat(checkIndexUpgraded.getKey(), equalTo(200));
+            assertThat(checkIndexUpgraded.getValue(), containsString(originalIndexName));
+
+
+            upgradedSourceOperations.deleteAllSnapshotsAndRepository(snapshotCreator.getRepoName());
```

Comment:
Do we need this function? IMO is OK to have more than one snapshot repo in a cluster